src/dispensary.js

```diff
@@ -65,13 +67,36 @@ export default class Dispensary {
       });
   }
 
-  getLibraries() {
+  // Matches only against cached hashes; this is the API external apps and
+  // libraries would use.
+  match(contents, _hashesFile=DEFAULT_HAHES_FILE, _cachedHashes=null) {
+    if (this._cachedHashes === null) {
+      if (_cachedHashes !== null) {
+        this._cachedHashes = _cachedHashes;
+      } else {
+        this._cachedHashes = this._getCachedHashes(_hashesFile);
+      }
+    }
+
+    for (let hashEntry of this._cachedHashes) {
+      let hash = hashEntry.split(' ')[0];
+      let library = hashEntry.split(' ')[1];
+
+      if (hasher(contents) === hash) {
```

Comment:
Does `hasher()` create a hash? Perhaps `createHash` or just `hash` might make more sense for this function name (should have caught this earlier)